design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+
+- Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+- Define the task functions of the ClusterLink operator.
+- Define the CRD (Custom Resource Definition) for the ClusterLink operator.
+
+## Non-Goals
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+- This document focuses only on the deployment of ClusterLink to the K8s cluster. 
+The deployment to another environment as VMs is out of scope. 
+- The security aspects and user permissions concerning the creation of certificates, specifically for fabric or peer.
+## Proposal
+The ClusterLink deployment will be done in the following steps, (as described in more detail in the User section):
+
+1. Create a Fabric certificate if it does not exist.
+2. Create the ClusterLink operator.
+3. Deploy the ClusterLink CRD to the operator, which will create peer certificates, ClusterLink deployments (control plane and data plane), services, and external access point.
+
+### ClusterLink CRD
+The ClusterLink CRD includes the following fields:
+- Api version: clusterlink.net/v1alpha1
+- Kind: clusterlink
+
+- **Spec Section:**
+  
+    | Field name | Description | Default value |
+    | ---- | ----- | ------ |
+    |dataplaneType| Types of dataplane, supported values "go" or "envoy"|"envoy"|
+    |dataplaneReplicates| Number of dataplane replicas|1|
+    |logLevel| Log level severity| "info"|
+    |logFile| Path to a file where logs will be written| By default logs will be printed to stderr|
+    |image| The container registry to pull the project images. If empty, it will use the local registry| ghcr.io/clusterlink-net.
+    |serviceExposureType| Type of service to expose ClusterLink deployment, supported values: "load-balancer", "API-gateway", "node-port". |load-balancer|
+
+- **Status section:**
+
+    Todo
+## Impacts / Key Questions
+
+- Do we need the ClusterLink CLI? The only command that seems necessary is for fabric and peer certificates.
+- We don't use or initialize gwctl anymore.
+- Need to have a security discussion focusing on how and who deploys the peer and fabric certificates. 
+
+## Risks and Mitigations
+
+Not applicable.
+
+### Security Considerations
+
+The assumption in this design document is that there isn't a distinction in the security level between creating fabric certificates and peer certificates. Users with permission for one are assumed to have permission for the other.  
+Additionally, the Fabric certificate should be accessible to the k8s operator by passing through k8s secrets during the creation of peer certificates.
+
+## Future Milestones
+In the first step, the k8s operator will be built and will utilize the current cl-adm implementation to create peer and fabric certificates.  
+The `cl-adm create peer1` command will also generate the files: k8s-secret.yaml (containing all the certificates for the control-plane and data-plane) and clusterlink.yaml (containing the manifests for the cl-operator).  
```

Comment:
do you propose to keep the `cl-adm` code or executable? That is, keeping the executable with all subcommands or just the package that creates the needed certificates.

Reply:
It may be confusing to keep `cl-adm`